internal/services/cdn/frontdoorruleactions/cdn_frontdoor_rule_actions.go

```diff
@@ -207,49 +208,104 @@ func ExpandCdnFrontDoorUrlRewriteAction(input []interface{}) (*[]cdn.BasicDelive
+		// however, if you do not define the Origin Group ID you also cannot define the Forwarding Protocol either
+		if originGroupIdRaw != "" {
+			originGroupOverride = cdn.OriginGroupOverride{
+				OriginGroup: &cdn.ResourceReference{
+					ID: utils.String(originGroupIdRaw),
+				},
+				ForwardingProtocol: forwardingProtocol,
+			}
+		} else if originGroupIdRaw == "" && item["forwarding_protocol"].(string) != "" {
+			return nil, fmt.Errorf("the 'route_configuration_override_action' block is not valid, if the 'cdn_frontdoor_origin_group_id' is not set you cannot define the 'forwarding_protocol', got %q", forwardingProtocol)
 		}
 
-		queryStringCachingBehavior := cacheConfiguration.QueryStringCachingBehavior
-		if queryParameters := cacheConfiguration.QueryParameters; queryParameters == nil {
-			if queryStringCachingBehavior == cdn.RuleQueryStringCachingBehaviorIncludeSpecifiedQueryStrings || queryStringCachingBehavior == cdn.RuleQueryStringCachingBehaviorIgnoreSpecifiedQueryStrings {
-				return nil, fmt.Errorf("the 'route_configuration_override_action' block is not valid, 'query_string_parameters' can not be empty if the 'query_string_caching_behavior' is set to 'IncludeSpecifiedQueryStrings' or 'IgnoreSpecifiedQueryStrings'")
+		if cacheBehavior == string(cdn.RuleIsCompressionEnabledDisabled) {
+			if queryStringCachingBehavior := cdn.RuleQueryStringCachingBehavior(item["query_string_caching_behavior"].(string)); queryStringCachingBehavior != "" {
+				return nil, fmt.Errorf("the 'route_configuration_override_action' block is not valid, if the 'cache_behavior' is set to 'Disabled' you cannot define the 'query_string_caching_behavior', got %q", queryStringCachingBehavior)
+			}
+
+			if queryParameters := item["query_string_parameters"].([]interface{}); len(queryParameters) != 0 {
+				return nil, fmt.Errorf("the 'route_configuration_override_action' block is not valid, if the 'cache_behavior' is set to 'Disabled' you cannot define the 'query_string_parameters', got %d", len(queryParameters))
+			}
+
+			if cacheDuration := item["cache_duration"].(string); cacheDuration != "" {
+				return nil, fmt.Errorf("the 'route_configuration_override_action' block is not valid, if the 'cache_behavior' is set to 'Disabled' you cannot define the 'cache_duration', got %q", cacheDuration)
 			}
 		} else {
-			if queryStringCachingBehavior == cdn.RuleQueryStringCachingBehaviorUseQueryString || queryStringCachingBehavior == cdn.RuleQueryStringCachingBehaviorIgnoreQueryString {
-				return nil, fmt.Errorf("the 'route_configuration_override_action' block is not valid, 'query_string_parameters' must not be set if the'query_string_caching_behavior' is set to 'UseQueryStrings' or 'IgnoreQueryStrings'")
+			// since 'cache_duration', 'query_string_caching_behavior' and 'cache_behavior' are optional create a default values
+			// for those values if not set.
+			cacheDuration := "1.12:00:00"
```

Comment:
is there some context for this default value, presumably this is coming from the API's default value/the portal or something?

Reply:
There is, Front Door will default the cache to somewhere between 1 to 3 days, I have no way of knowing what Front Door decided and since I had to offer a default value to match the portal experience and this field has to be optional now I picked a value in the middle of what the Front Door documentation says the cache range can be... it seemed to me to be the lesser of the two evils and also allows for parity in functionality between Terraform and the Portal, which does not currently exist in any other CLI solutions.